### PR TITLE
TS errors cleaned up

### PR DESCRIPTION
**Pull request by gatinhap**

### CODE REVIEW

File: src/pages/removePlantPage/RemovePlant.tsx [hunk 1]
```diff
@@ -13,16 +13,16 @@ import { toast } from "react-toastify";
...
 const RemovePlant = () => {
-  const { plantId } = useParams();
+  const { plantId } = useParams<{ plantId: string }>();
   const navigateTo = useNavigate();
   const queryClient = useQueryClient();
 
-  const removePlant = (id) => {
-    return pb.collection(PLANTS_COLLECTION).delete(id);
+  const removePlant = (id: string) => {
+    return pb.collection(PLANTS_COLLECTION).delete(id as string);
```
Comment (michal-karbasz) [inline src/pages/removePlantPage/RemovePlant.tsx:21]: hm why do you need type assertions here and below? You have just defined id as string. Also it seems you can just use the plantId from useParams?

File: src/components/userForms/UserRegistrationForm.tsx [hunk 2]
```diff
@@ -64,12 +59,18 @@ const UserRegistrationForm = () => {
     },
   });
 
+  const submitUserData: SubmitHandler<UserRegistrationFormValues> = (
+    userData,
+  ) => {
+    return addUserMutation.mutate(userData);
```
Comment (michal-karbasz) [inline src/components/userForms/UserRegistrationForm.tsx:65]: redundant return statement and brackets

File: src/components/userForms/LoginForm.tsx [hunk 3]
```diff
@@ -48,11 +41,18 @@ const LoginForm = () => {
     },
   });
 
+  const submitUserLoginData: SubmitHandler<LoginFormValues> = ({
+    email,
+    password,
+  }) => {
```
Comment (michal-karbasz) [inline src/components/userForms/LoginForm.tsx:47]: redundant return statement, just use => please

File: src/pages/removePlantPage/RemovePlant.tsx (hunk 1 above)
Comment (gatinhap) [reply src/pages/removePlantPage/RemovePlant.tsx:21]: in line 20 if I remove type assertion, TS throws error that id has implicitly type any
in line 21, I removed it

**Review by michal-karbasz — APPROVED** [review]

**Pull request merged**